apps/hello/main.py

```diff
@@ -126,11 +138,34 @@ def _update_scatter_chart(state):
     )
     state["scatter_chart"] = fig
 
-# STATE INIT
 
+# UPDATES DATAFRAME
+
+# Subscribe this event handler to the `wf-dataframe-add` event
+def on_editable_df_record_add(state, payload):
+	state['editable_df'].record_add(payload)
+
+
+# Subscribe this event handler to the `wf-dataframe-update` event
+def on_editable_df_record_change(state, payload):
+    payload['record']['object']['updatedAt'] = str(datetime.datetime.now())
+    state['editable_df'].record_update(payload)
+
+# Subscribe this event handler to the `wf-dataframe-action` event
+def on_editable_df_record_action(state, payload):
+	record_index = payload['record_index']
+	if payload['action'] == 'remove':
+		state['editable_df'].record_remove(payload)
+	if payload['action'] == 'open':
+		state['editable_df_open_text'] = str(state['editable_df'].record(record_index)['text'])
+
```

Comment:
Here is a suggestion for the `test` action. 

![image](https://github.com/user-attachments/assets/721b2b51-88a5-48e7-aee9-67596e775453)

```suggestion
# Subscribe this event handler to the `wf-dataframe-action` event
def on_editable_df_record_action(state, payload):
    record_index = payload['record_index']
    if payload['action'] == 'remove':
        state['editable_df'].record_remove(payload)
    if payload['action'] == 'open':
        state['editable_df_open_text'] = str(state['editable_df'].record(record_index)['text'])
    if payload['action'] == 'test':
        state.add_notification("info", "Test", "select : " + str(state['editable_df'].record(record_index)))
```

Reply:
Good idea, implemented